File: 2025/3.js
const input = [
  987654321111111,
  811111111111119,
  234234234234278,
  818181911112111,
];

function findBiggestNumber(input, high) {
  let largestNumber = 0;

  high ??= input.length;

  for (let index = 0; index < high; index++) {
    if (input[index] > largestNumber) largestNumber = String(input[index]);
  }

  return largestNumber;
}

let total = 0;

for (let item of input) {
  item = String(item);

  const firstValue = findBiggestNumber(item, item.length - 1);
  const indexOfFirstValue = String(item).indexOf(firstValue);

  const secondValue = findBiggestNumber(
    item.slice(indexOfFirstValue + 1),
  );

  total += Number(`${firstValue}${secondValue}`);
}

console.log(total);

File: 2025/3.go
package main

import (
	"fmt"
	"log"
	"strconv"
	"strings"
)

func findBiggestNumber(input string, high int) (int, error) {
	largestNumber := 0

	inputSlice := []rune(input)

	for i := 0; i < high; i++ {
		item, err := strconv.Atoi(string(inputSlice[i]))

		if err != nil {
			return -1, err
		}

		if item > largestNumber {
			largestNumber = item
		}
	}

	return largestNumber, nil
}

func main() {
	input := [...]int{
		987654321111111,
		811111111111119,
		234234234234278,
		818181911112111,
	}

	log.Fatal("ooo")
	total := 0

	for _, value := range input {
		stringifiedValue := strconv.Itoa(value)

		firstValue, err := findBiggestNumber(stringifiedValue, len(stringifiedValue)-1)
		if err != nil {
			log.Fatal(err.Error())
		}

		slices := strings.Split(stringifiedValue, strconv.Itoa(firstValue))
		relevantSlice := slices[1]

		secondValue, err := findBiggestNumber(relevantSlice, len(relevantSlice))
		if err != nil {
			log.Fatal(err.Error())
		}

		stringTotal := strconv.Itoa(firstValue) + strconv.Itoa(secondValue)

		intTotal, err := strconv.Atoi(stringTotal)

		total += intTotal
	}

	fmt.Println(total)
}
